### PR TITLE
Fix FrozenCollection tests failing in Native AOT

### LINKED ISSUE #88628: Frozen collection tests fail on NativeAOT
LookupItems_AllItemsFoundAsExpected test is failing on `runtime-extra-platforms`.

https://dev.azure.com/dnceng-public/public/_build/results?buildId=333900&view=results

```
System.NotSupportedException : 'Xunit.Sdk.AssertEqualityComparer1[System.Collections.Generic.HashSet1[System.Collections.Generic.KeyValuePair2[System.UIntPtr,System.UIntPtr]]].CompareTypedSets[System.Collections.Generic.KeyValuePair2[System.UIntPtr,System.UIntPtr]](https://dev.azure.com/dnceng-public/public/_build/results?buildId=333900&view=ms.vss-test-web.build-test-results-tab&runId=6918600&resultId=106768&paneView=debug)' is missing native code. MethodInfo.MakeGenericMethod() is not compatible with AOT compilation. Inspect and fix AOT related warnings that were generated when the app was published. For more information see https://aka.ms/nativeaot-compatibility

```

```
Stack trace
   at System.Reflection.Runtime.MethodInfos.RuntimeNamedMethodInfo`1.GetUncachedMethodInvoker(RuntimeTypeInfo[], MemberInfo) + 0x74
   at System.Reflection.Runtime.MethodInfos.RuntimeNamedMethodInfo`1.MakeGenericMethod(Type[]) + 0x1ff
   at System.Collections.Frozen.Tests.FrozenDictionary_Generic_Tests`2.LookupItems_AllItemsFoundAsExpected(Int32 size, IEqualityComparer`1 comparer, Boolean specifySameComparer) + 0x296
   at System.Collections.Immutable.Tests!<BaseAddress>+0xc2051e
   at System.Reflection.DynamicInvokeInfo.Invoke(Object, IntPtr, Object[], BinderBundle, Boolean) + 0x167
```

This test failed in https://github.com/dotnet/runtime/issues/78046 for the same reason. To fix that issue, we added an rd.xml file to ensure `Int32` native code was generated during PublishAot.

However, with #88093, we now added more primitive types to the test - see System.UIntPtr above.

To fix this, we need to add more types to the rd.xml file that was added in https://github.com/dotnet/runtime/pull/78005.

https://github.com/dotnet/runtime/blob/e40d6e165c10fe62cf348405ffcc6c39e96127b2/src/libraries/System.Collections.Immutable/tests/default.rd.xml#L34-L38

cc @adamsitnik @VSadov

### PR DESCRIPTION
**Pull request by eiriktsarpalis**
Fix #88628.

### CODE REVIEW

File: src/libraries/System.Collections.Immutable/tests/default.rd.xml [hunk 1]
```diff
@@ -37,24 +37,108 @@
         </Method>
       </Type>
 
+      <Type Name="Xunit.Sdk.AssertEqualityComparer`1[[System.Collections.Generic.HashSet`1[[System.Collections.Generic.KeyValuePair`2[[System.SByte, System.Private.CoreLib],[System.SByte, System.Private.CoreLib]],System.Private.CoreLib]],System.Private.CoreLib]]" Dynamic="Required All">
```
Comment (eerhardt) [inline src/libraries/System.Collections.Immutable/tests/default.rd.xml:40]: @agocke, do you have a timeline when we will have Xunit asserts that work in NativeAOT?

**Review by eerhardt — APPROVED** [review]
Assuming the tests pass now - LGTM

File: src/libraries/System.Collections.Immutable/tests/default.rd.xml (hunk 1 above)
Comment (agocke) [reply src/libraries/System.Collections.Immutable/tests/default.rd.xml:40]: I have it in a branch locally, and just keep getting interrupted by other stuff. I will likely work on it more after we shut down the platform, as infra will be more important then.

**Pull request merged**